compiler/rustc_data_structures/src/profiling.rs

```diff
@@ -393,7 +393,7 @@ impl SelfProfilerRef {
     }
 
     /// Record a query in-memory cache hit.
-    #[inline(always)]
+    #[inline(never)]
     pub fn query_cache_hit(&self, query_invocation_id: QueryInvocationId) {
```

Comment:
The same should probably be done for the other profiling events as well

Reply:
The problem was that it generated code for `TimingGuard` which was unused. The other events do make use of it so it outlining doesn't help.